### PR TITLE
refactor(useLocale): make component locale overriding locale from con…

### PR DESCRIPTION
**Pull request by Asanio06**
…text

[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

Today, if I inject parameters into the `locale` props in my Calendar component, it doesn't take them into account if I already have a `locale` in my ConfigProvider. This means that I can't have a global configuration that I override for specific cases. Here some exemple : https://stackblitz.com/edit/react-o7hv5j?file=demo.tsx 
You can see that if you remove ConfigProvider locale, all we work fine

### 📝 Changelog

- Use a developer-oriented tone and narrative style.
- Describe the user's first-hand experience of the issue and its impact on developers, rather than your solution approach.
- Refer to: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     refactor(useLocale): make component locale overriding locale from context      |
| 🇨🇳 Chinese |           |

### CODE REVIEW

**Comment by afc163** [comment]
Please create a online reproduce and describe what problem this PR you want to resolve?

**Review by afc163 — CHANGES_REQUESTED** [review]
Please create a online reproduce and describe what problem this PR you want to resolve?

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "antd",
-  "version": "5.19.4",
+  "version": "5.20.0",
```
Comment (li-jia-nan) [inline package.json:3]: Please do not modify the version

**Comment by Asanio06** [comment]
Sorry i will make all change you talk about

File: package.json (hunk 1 above)
Comment (Asanio06) [reply package.json]: Sorry, i have change that

**Comment by Asanio06** [comment]
Hello @afc163 , I've updated my code to be clearer and not to cause any breaking changes.
Here some exemple : https://stackblitz.com/edit/react-o7hv5j?file=demo.tsx
You can see that if you remove ConfigProvider locale, all we work fine

**Pull request closed without merging**